BlockBlobReader/target/consumer_build/BlobTaskConsumer/index.js

```diff
@@ -424,68 +327,81 @@ function servicebushandler(context, serviceBusTask) {
     }
 
     sumoClient = new sumoHttp.SumoClient(options, context, failureHandler, successHandler);
+    context.log("Reached inside ServiceBus Handler")
     messageHandler(serviceBusTask, context, sumoClient);
 
 }
 
-function timetriggerhandler(context, timetrigger) {
+async function timetriggerhandler(context, timetrigger) {
 
     if (timetrigger.isPastDue) {
         context.log("timetriggerhandler running late");
     }
-    var serviceBusService = servicebus.createServiceBusService(process.env.APPSETTING_TaskQueueConnectionString);
-    serviceBusService.receiveQueueMessage(process.env.APPSETTING_TASKQUEUE_NAME + '/$DeadLetterQueue', {isPeekLock: true}, function (error, lockedMessage) {
-        if (!error) {
-            var serviceBusTask = JSON.parse(lockedMessage.body);
-            // Message received and locked and try to resend
-            var options = {
-                urlString: process.env.APPSETTING_SumoLogEndpoint,
-                MaxAttempts: 3,
-                RetryInterval: 3000,
-                compress_data: true,
-                clientHeader: "dlqblobreader-azure-function"
-            };
-            setSourceCategory(serviceBusTask, options);
-            var sumoClient;
-            function failureHandler(msgArray, ctx) {
-                ctx.log("Failed to send to Sumo");
-                if (sumoClient.messagesAttempted === sumoClient.messagesReceived) {
-                    ctx.done("TaskConsumer failedmessages: " + sumoClient.messagesFailed);
-                }
+
+    var sbClient = new ServiceBusClient(process.env.APPSETTING_TaskQueueConnectionString);
+    var queueReceiver = sbClient.createReceiver(process.env.APPSETTING_TASKQUEUE_NAME + '/$DeadLetterQueue');
+
+    try {
+        var message = await queueReceiver.receiveMessages(1, {
+            maxWaitTimeInMs: 60 * 1000,
```

Comment:
also set peek lock to true so that that message is not visible to  any other function

Reply:
peeklock is implemeted in receiving streaming use [case](https://github.com/Azure/azure-sdk-for-js/blob/b6ec7571e28576a5411fb452715c4be957ce2a2c/sdk/servicebus/service-bus/samples/v7/javascript/receiveMessagesStreaming.js#L34)

Reply:
array use case doesn't seem to have a peek lock use case implemented